src/plugins/tscn/tscnplugin.cpp

```diff
@@ -182,6 +190,52 @@ static void findUsedTilesets(const TileLayer *layer, AssetInfo &assetInfo)
     }
 }
 
+// Search an object layer for all object resources and save them in assetInfo
+static void findUsedObjects(const ObjectGroup *objectLayer, AssetInfo &assetInfo)
+{
+    for (const MapObject *object : objectLayer->objects()) {
+        auto resPath = object->resolvedProperty("resPath").toString();
+
+        if (resPath.isEmpty()) {            
+            if (resPath.isEmpty()) {
+                Tiled::WARNING(TscnPlugin::tr("Only objects with the resPath property will be exported"),
+                    Tiled::JumpToObject { object });
+                continue;
+            }
+        }
+        
+        QRegularExpressionMatch match;
+        if (!resPath.contains(QRegularExpression("^res://(.*)\\.tscn$"), &match)) {
+            Tiled::ERROR(TscnPlugin::tr("resPath must be in the form of 'res://<filename>.tscn'."),
+                Tiled::JumpToObject { object });
+            continue;
+        }
+        
+        QString baseName = sanitizeSpecialChars(match.captured(1));
+        int uniqueifier = 1;
```

Comment:
Since the object ID should always be unique for a given map, would it make sense to use it instead of an incremental uniqueifier?

Reply:
The ID should be unique per res path, not per object. I've updated the comments to reflect that.

Reply:
I made this change in the last commit. I could take it or leave it. The pros:

* This reduces the line found from 14 to 8.
* This is more efficient.

The cons:

* The generated files are less human readable.
* It makes the files less git-friendly. For instance, given:

```
[node name="Object8" parent="." instance=ExtResource("object_resource_0")]
position = Vector2(320, 10)

[node name="Object9" parent="." instance=ExtResource("object_resource_1")]
position = Vector2(125, 29.5)

[node name="Object10" parent="." instance=ExtResource("object_resource_2")]
position = Vector2(172, 29.5)
```

If you delete the first object from the map, then every object afterwards will change, because object_resource_1 will be renamed to object_resource_0, etc.

With the old code, it would remove Object8 and leave it at that.

Reply:
I liked the simplification, but I think you're concerns are valid and likely the previous approach will be preferred by some.